subsys/bluetooth/mesh/proxy_msg.h

```diff
@@ -27,6 +27,8 @@
 
 struct bt_mesh_proxy_role;
 
+extern int bt_mesh_conn_count;
+
```

Comment:
Add a getter function for this and make the variable static and you will be guaranteed that proxy_msg.c is the only one modifying it.